CONTRIBUTING.md

```diff
@@ -5,70 +5,107 @@ If you have any questions or suggestions we are happy to hear them.
 exist.  
-* If it exists the Quarkus language server sends them to VSCode 
+* If it exists the MicroProfile language server sends them to VSCode 
 as the response to the `textDocument/completion` request.
 Communication is complete, and does not proceed to Step 3 
 and onwards.
-* If it does not exist, the Quarkus language server sends a 
-custom request, `quarkus/projectInfo` to vscode-quarkus.  
+* If it does not exist, the MicroProfile language server sends a 
+custom request, `microprofile/projectInfo` to vscode-quarkus.  
 Proceed to Step 3.  
 
-Step 3. vscode-quarkus receives the `quarkus/projectInfo` request,
-and runs the `quarkus.java.projectInfo` command. This command was
-defined by the Quarkus jdt.ls extension.  
+**Step 3.** vscode-quarkus receives the `microprofile/projectInfo` request,
+and delegates it to the MicroProfile jdt.ls extension.  
 
-Step 4. The Quarkus jdt.ls extension receives the command, determines
-information about the currently opened Quarkus project and returns
-the information to vscode-quarkus.  
+**Step 4.** The MicroProfile jdt.ls extension receives the command, determines
+[project information](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/commons/MicroProfileProjectInfo.java)
+(project URI, configuration properties, hints etc.)
+about the currently opened Quarkus project and returns
+the information to vscode-quarkus. The project information is then sent to
+the MicroProfile language server
 
-The information is computed by scanning all classes annotated with `@ConfigRoot` Quarkus annotation from all project classpath JARs and generating the proper Quarkus properties.
+**Step 5.** vscode-quarkus receives the project information and sends it
+to the MicroProfile language server.  
 
-For instance if the project has a `pom.xml` with this dependency:
+**Step 6.** MicroProfile language server receives the information, adds it 
+to its cache, and returns the completion options stored in the 
+project information as the response to the `textDocument/completion`
+request.
 
-```xml
-<dependency>
-    <groupId>io.quarkus</groupId>
-    <artifactId>quarkus-core-deployment</artifactId>
-</dependency>
-```
+## Implementing language features for `application.properties`
+When an `application.properties` file sends a request (e.g. textDocument/completion) to the
+MicroProfile language server, the requests are accepted in
+[ApplicationPropertiesTextDocumentService#completion](https://github.com/redhat-developer/quarkus-ls/blob/9eb7718a6e0faa300f3937e607c3bfffc25c2f1d/microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/ls/ApplicationPropertiesTextDocumentService.java#L150).
+This class receives LSP requests for `application.properties` files.
+
+The completion items are being determined from the MicroProfile jdt.ls extension, which searches Quarkus project
```

Comment:
When the completion is triggered, it search in their MicroProfile LS cache the properties for the given application.properties. If cache is not filled, it call the `microprofile/projectInfo` request to call the JDT LS Extension [projectInfo delegate command handler ](https://github.com/redhat-developer/quarkus-ls/blob/0ff91b4d0fe4670584a3ad23fe645c8141df7f3d/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/ls/MicroProfileDelegateCommandHandler.java#L99) in otherwise the manager which collect MicroProfile, Quarkus properties for a given java project.

This manager is extensible by the extension point `com.redhat.microprofile.jdt.core.propertiesProviders`:

 * [com.redhat.microprofile.jdt.core](https://github.com/redhat-developer/quarkus-ls/blob/0ff91b4d0fe4670584a3ad23fe645c8141df7f3d/microprofile.jdt/com.redhat.microprofile.jdt.core/plugin.xml#L33) define properties provider for MicroProfile.
 * [https://github.com/redhat-developer/quarkus-ls/blob/0ff91b4d0fe4670584a3ad23fe645c8141df7f3d/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/plugin.xml#L5) define properties provider for Quarkus.

Reply:
How's this?
![image](https://user-images.githubusercontent.com/20326645/74756555-bbc83200-5242-11ea-9a83-b6bff4d90415.png)